### PR TITLE
faux_generator

### PR DESCRIPTION
**Pull request by ldjebran**
close https://github.com/omaciel/pytest-fauxfactory/issues/5

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -164,6 +163,115 @@ This will generate 5 new tests
...
+
+::
+
+    tests/test_pytest_fauxfactory.py::test_generator_expression[7567941589364906197] PASSED
+    tests/test_pytest_fauxfactory.py::test_generator_expression[5816026928693750395] PASSED
+    tests/test_pytest_fauxfactory.py::test_generator_expression[7144149572714817589] PASSED
+    tests/test_pytest_fauxfactory.py::test_generator_expression[5753372709616898246] PASSED
+
+
+of cause the returned values can be of any type:
```
Comment (omaciel) [inline README.rst]: Of course

File: README.rst [hunk 2]
```diff
@@ -164,6 +163,115 @@ This will generate 5 new tests
...
 
-   tests/test_pytest_fauxfactory.py::test_generate_person[value0] PASSED
-   tests/test_pytest_fauxfactory.py::test_generate_person[value1] PASSED
-   tests/test_pytest_fauxfactory.py::test_generate_person[value2] PASSED
+    tests/test_pytest_fauxfactory.py::test_generate_person[value0] PASSED
+    tests/test_pytest_fauxfactory.py::test_generate_person[value1] PASSED
+    tests/test_pytest_fauxfactory.py::test_generate_person[value2] PASSED
+
+Now instead of using a callable function, we want to generate tests with values
+of any types from a generator function or generator expression,
```
Comment (omaciel) [inline README.rst]:
```diff
-of any types from a generator function or generator expression,
-for this purpose we use the "faux_generator" mark:
+of any types from a generator function or generator expression.
+For this purpose we can use the "faux_generator" mark:
```

File: README.rst [hunk 3]
```diff
@@ -164,6 +163,115 @@ This will generate 5 new tests
...
+
+
+This will generate 2 tests
+
+::
+
+    tests/test_pytest_fauxfactory.py::test_generator_foo_generator[foo] PASSED
+    tests/test_pytest_fauxfactory.py::test_generator_foo_generator[value1] PASSED
+
+we can also combine all the above generators:
```
Comment (omaciel) [inline README.rst]: s/we/We

File: pytest_fauxfactory.py [hunk 4]
```diff
@@ -68,6 +76,25 @@ def _pytest_faux_callable_mark_handler(metafunc):
     return faux_callable(items, callable_function, *args[2:], **kwargs)
 
 
+def _pytest_faux_generator_mark_handler(metafunc):
+    """"pytest faux generator mark handler"""
```
Comment (omaciel) [inline pytest_fauxfactory.py]: s/faux generator/faux_generator

File: tests/test_pytest_fauxfactory.py [hunk 5]
```diff
@@ -304,3 +304,124 @@ def test_callable_generate_person(value):
...
+            assert value
+    """)
+    result = testdir.runpytest()
+    result.assert_outcomes(error=1)
+    assert 'Argument with index 1 is not a generator' in result.stdout.str()
+    assert result.ret == 2
+
+
+def test_generator_mark_number_of_tests_generated(testdir):
+    """check that the number of tests generated is correct."""
```
Comment (omaciel) [inline tests/test_pytest_fauxfactory.py]: s/check/Check

File: tests/test_pytest_fauxfactory.py [hunk 6]
```diff
@@ -304,3 +304,124 @@ def test_callable_generate_person(value):
...
+        def test_generator_test_number(value):
+            assert value
+    """)
+    result = testdir.runpytest()
+    result.assert_outcomes(passed=10)
+    assert result.ret == 0
+
+
+def test_generator_mark_combined_gens_number_of_tests_generated(testdir):
+    """check that using combined generators, generates the correct number of
```
Comment (omaciel) [inline tests/test_pytest_fauxfactory.py]:
```diff
-check that using combined generators, generates the correct number of
+Check that using combined generators generates the correct number of
```

File: tests/test_pytest_fauxfactory.py [hunk 7]
```diff
@@ -304,3 +304,124 @@ def test_callable_generate_person(value):
...
+    """Test function generator with kwargs"""
+    assert len(value) == 12
+
+
+list_of_integers = [fauxfactory.gen_integer(min_value=0) for _ in range(4)]
+
+
+@pytest.mark.faux_generator(int_val for int_val in list_of_integers)
+def test_generator_expression(value):
+    """Test generator expression"""
```
Comment (omaciel) [inline tests/test_pytest_fauxfactory.py]: Missing period at the end.

File: tests/test_pytest_fauxfactory.py [hunk 8]
```diff
@@ -304,3 +304,124 @@ def test_callable_generate_person(value):
...
+def foo_generator():
+    """Returns different values: first, a string 'foo'; second iteration, a
+    list of integers."""
+    yield 'foo'
+    yield [1, 2, 3]
+
+
+@pytest.mark.faux_generator(foo_generator())
+def test_generator_foo_generator(value):
+    """Test diffrent types values"""
```
Comment (omaciel) [inline tests/test_pytest_fauxfactory.py]:
```diff
-Test diffrent types values
+Test different type values.
```

File: tests/test_pytest_fauxfactory.py [hunk 9]
```diff
@@ -304,3 +304,124 @@ def test_callable_generate_person(value):
...
+        assert value == 'foo'
+
+
+@pytest.mark.faux_generator(
+    alpha_strings_generator(items=3, length=12),
+    (int_val for int_val in list_of_integers),
+    foo_generator()
+)
+def test_generator_combined(value):
+    """Test combined generators"""
```
Comment (omaciel) [inline tests/test_pytest_fauxfactory.py]: Missing period at the end.

File: pytest_fauxfactory.py (hunk 4 above)
Comment (omaciel) [reply pytest_fauxfactory.py]: Missing period at the end.

File: tests/test_pytest_fauxfactory.py [hunk 10]
```diff
@@ -304,3 +304,124 @@ def test_callable_generate_person(value):
...
+
+def alpha_strings_generator(items=1, length=10):
+    """Generate alpha string value at each iteration."""
+    for _ in range(items):
+        yield fauxfactory.gen_alpha(length=length)
+
+
+@pytest.mark.faux_generator(alpha_strings_generator(items=3, length=12))
+def test_generator_alpha_strings(value):
+    """Test function generator with kwargs"""
```
Comment (omaciel) [inline tests/test_pytest_fauxfactory.py]: Missing period at the end.

**Comment by ldjebran** [comment]
@omaciel changes done

**Comment by omaciel** [comment]
./pytest_fauxfactory.py:3:1: F401 'inspect.isgeneratorfunction' imported but unused

**Comment by ldjebran** [comment]
@omaciel done

**Pull request merged**